cmd/otel-allocator/collector/collector_test.go

```diff
@@ -0,0 +1,96 @@
+package collector
+
+import (
+	"context"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/client-go/kubernetes/fake"
+)
+
+func Test(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	collectors := []string{}
+
+	client := Client{
+		k8sClient:     fake.NewSimpleClientset(),
+		collectorChan: make(chan []string),
+	}
+
+	labelMap := map[string]string{
+		"app.kubernetes.io/instance":   "default.test",
+		"app.kubernetes.io/managed-by": "opentelemetry-operator",
+	}
+
+	opts := metav1.ListOptions{
+		LabelSelector: labels.SelectorFromSet(labelMap).String(),
+	}
+
+	client.Watch(ctx, labelMap, func(collectorList []string) {})
+	// adding sleep to alow the collector watch function to start
+	time.Sleep(1 * time.Second)
+
+	t.Run("should create pods", func(t *testing.T) {
+		expected := pod("test-pod1")
+		_, err := client.k8sClient.CoreV1().Pods("test-ns").Create(ctx, expected, metav1.CreateOptions{})
+		assert.NoError(t, err)
+
+		collectors = <-client.collectorChan
```

Comment:
Where are you using this?

Reply:
This was added just to start with an one collector instance in the cluster. I have removed this as it is not required and it can be misleading as we dont have the functionality of adding a pod.